sale_timesheet_task_exclude/tests/test_sale_timesheet_exclude_task.py

```diff
@@ -0,0 +1,129 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo.tests import common
+
+
+class TestSaleTimesheetExcludeTask(common.TransactionCase):
+
+    def setUp(self):
+        super().setUp()
+
+        self.uom_hour = self.env.ref('uom.product_uom_hour')
+        self.user_type_payable = self.env.ref(
+            'account.data_account_type_payable'
+        )
+        self.user_type_receivable = self.env.ref(
+            'account.data_account_type_receivable'
+        )
+        self.user_type_revenue = self.env.ref(
+            'account.data_account_type_revenue'
+        )
+        self.Partner = self.env['res.partner']
+        self.SudoPartner = self.Partner.sudo()
+        self.Employee = self.env['hr.employee']
+        self.SudoEmployee = self.Employee.sudo()
+        self.AccountAccount = self.env['account.account']
+        self.SudoAccountAccount = self.AccountAccount.sudo()
+        self.Project = self.env['project.project']
+        self.SudoProject = self.Project.sudo()
+        self.ProjectTask = self.env['project.task']
+        self.SudoProjectTask = self.ProjectTask.sudo()
+        self.AccountAnalyticLine = self.env['account.analytic.line']
+        self.SudoAccountAnalyticLine = self.AccountAnalyticLine.sudo()
+        self.ProductProduct = self.env['product.product']
+        self.SudoProductProduct = self.ProductProduct.sudo()
+        self.SaleOrder = self.env['sale.order']
+        self.SudoSaleOrder = self.SaleOrder.sudo()
+        self.SaleOrderLine = self.env['sale.order.line']
+        self.SudoSaleOrderLine = self.SaleOrderLine.sudo()
+        self.ProjectCreateSaleOrder = self.env['project.create.sale.order']
+
+    def test_1(self):
+        account = self.SudoAccountAccount.create({
```

Comment:
Why don't you include on the setup all the preparation? I mean to create data is not what you want to test, right?

Reply:
Yes, but it's a data local to test-case, not test-suite.

Reply:
Sorry if not clear what is the purpose of my comment but I will try. Here you use Transaction test kind and just one test inside so that means you will not experience any problem but imagine you have three or more test and all of them needs this local data.

Each test will only share the setup and runs on it's own transaction so any change will not affect the others so normally you could add all the common data or setup on this constructor.

If you read my other comments I trying to convince you to split on more than one test with proper name so any other will easily understand how to maintain this module :)

cc @pedrobaeza @Yajo

Reply:
@victormmtorres when/if this would be needed for more than one test case (as it was before), it data will be either copied to complement local adjustments or will be shared if can be shared. At this point, I see no justification of which data to share or not, so such re-arrangement of code is premature